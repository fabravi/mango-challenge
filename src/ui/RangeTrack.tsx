import styles from "./rangetrack.module.scss";

export default function RangeTrack() {
  return <span className={styles.track}></span>;
}
