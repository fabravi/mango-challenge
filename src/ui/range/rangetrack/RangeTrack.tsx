import styles from "./rangetrack.module.scss";

interface RangeTrackProps {
  left: number;
  width: number;
}

export default function RangeTrack({ left, width }: RangeTrackProps) {
  return (
    <span
      role="slider-track"
      className={styles.track}
      style={{ left: `${left}%`, width: `${width}%` }}
    ></span>
  );
}
