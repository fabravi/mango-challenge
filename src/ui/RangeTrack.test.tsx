import { render } from "@testing-library/react";
import RangeTrack from "./RangeTrack";

describe("RangeTrack", () => {
  test("should render correctly", () => {
    const { container } = render(<RangeTrack left={0} width={100} />);
    expect(container).toMatchSnapshot();
  });

  test("should render the track with the correct left position and width", () => {
    const { container } = render(<RangeTrack left={50} width={50} />);
    const track = container.querySelector("span");
    expect(track).toHaveStyle("left: 50%");
    expect(track).toHaveStyle("width: 50%");
  });
});
