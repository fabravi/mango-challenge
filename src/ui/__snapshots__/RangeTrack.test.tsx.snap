// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeTrack should render correctly 1`] = `
<div>
  <span
    class="track"
    role="slider-track"
    style="left: 0%; width: 100%;"
  />
</div>
`;
